Hw10

tikalet opened the pull request.

tikalet committed:
add hw

tikalet committed:
add tests

vlad-iv commented on hw10/src/main/java/ru/otus/hw/controller/BookControllerRest.java:
```diff
@@ -0,0 +1,53 @@
+        return ResponseEntity.ok(bookService.findById(id));
+    }
+
+    @PostMapping("/api/book")
```
можно вернуть 201 код

vlad-iv commented on hw10/src/main/java/ru/otus/hw/controller/BookControllerRest.java:
```diff
@@ -0,0 +1,53 @@
+        return ResponseEntity.ok(bookDto);
+    }
+
+    @DeleteMapping("/api/book/{id}")
```
можно вернуть код 204

vlad-iv commented on hw10/src/main/java/ru/otus/hw/controller/BookControllerRest.java:
```diff
@@ -0,0 +1,53 @@
+    }
+
+    @DeleteMapping("/api/book/{id}")
+    public ResponseEntity<String> deleteBook(@PathVariable("id") Long id) {
```
Предлагаю вернуть void

vlad-iv commented on hw10/src/main/java/ru/otus/hw/exceptions/GlobalExceptionHandler.java:
```diff
@@ -0,0 +1,49 @@
+import java.util.stream.Collectors;
+
+@RequiredArgsConstructor
+@ControllerAdvice
```
Можно заменить на RestControllerAdvice

vlad-iv commented on hw10/src/main/java/ru/otus/hw/exceptions/GlobalExceptionHandler.java:
```diff
@@ -0,0 +1,49 @@
+
+    @ExceptionHandler(NotFoundException.class)
+    public ResponseEntity<ErrorDto> handeNotFoundException(NotFoundException ex) {
+        System.out.println("ERROR: " + ex.getMessage());
```
осталось заменить на логирование
@ Slfj4 над классом

vlad-iv commented on hw10/src/main/java/ru/otus/hw/exceptions/GlobalExceptionHandler.java:
```diff
@@ -0,0 +1,49 @@
+
+    @ExceptionHandler(Exception.class)
+    public ResponseEntity<ErrorDto> handeInternalError(Exception ex) {
+        System.out.println("ERROR: " + ex.toString());
```
предлагаю залогировать стек трейс

vlad-iv commented on hw10/src/main/java/ru/otus/hw/dto/BookCreateDto.java:
```diff
@@ -0,0 +1,28 @@
+public class BookCreateDto {
+
+    @Null
+    private Long id;
```
можно удалить поле

vlad-iv commented on hw10/src/main/java/ru/otus/hw/dto/CommentCreateDto.java:
```diff
@@ -0,0 +1,23 @@
+public class CommentCreateDto {
+
+    @Null
+    private Long id;
```
можно удалить поле

tikalet committed:
fix

- add log to ExceptionHandler
- fix css file name
- delete field from create bean
- fix code status
